tools/federation/src/main/java/org/eclipse/rdf4j/federated/FedXConfig.java

```diff
@@ -40,7 +40,7 @@ public class FedXConfig {
 
 	private int leftJoinWorkerThreads = 10;
 
-	private int boundJoinBlockSize = 15;
+	private int boundJoinBlockSize = 25;
```

Comment:
Does this kinda become the opposite of parallelism? As in if this variable is infinite then it would only use a single thread? 

Would it make sense to increase this variable at runtime? Start it off at 1 and then double it for every join until it reaches some kind of configurable max?

Reply:
> Does this kinda become the opposite of parallelism? As in if this variable is infinite then it would only use a single thread?

Exactly: the join operator blocks until a block is filled up prior to submitting it to the scheduler and executing it as sub-query.

So, this is really a tuning parameter of a federation. When FedX was developed some 10 years ago, in the benchmarks a value of 15 produced the best results. Nowadays it looks like higher values seem to produce better performance. In our scenarios we are currently looking for a value somwhere in the range 25 to 50.

> Would it make sense to increase this variable at runtime? Start it off at 1 and then double it for every join until it reaches some kind of configurable max?

Also this is the right observation. In fact this change is already doing a bit of preparation into this direction with moving the "getNextBindJoinSize" into a helper method. My idea is to do evaluations there first in specialized implementations. I think especially for large joins (with more than 1000 intermediate results) it pays off to increase the join size after the 1000 first results have been processed. I do not see this in 5.0 though, but a bit later once have more experience (or ideally objective benchmark results for some query mix)